pynamodb_attributes/datetime.py

```diff
@@ -0,0 +1,11 @@
+from pynamodb.attributes import Attribute
+from pynamodb.attributes import UTCDateTimeAttribute
+
+
+class DateTimeAttribute(Attribute):
+    """
+    Unlike UTCDateTimeAttribute, this attribute has its type hinted as 'datetime'.
```

Comment:
https://github.com/pynamodb/PynamoDB/blob/140993f453e7a52d99af5a9835b87aade648e47e/pynamodb/attributes.pyi#L121

How is this different?

Reply:
isn't the problem that `UTCDateTimeAttribute` _does_ have its type hinted as `datetime` and so the `set` method fails the mypy check?

Reply:
Could you give me an example of code that fails with plain PynamoDB and will be fixed by this?

Reply:
This is the code thats failing:
https://github.com/lyft/cancels/blob/d4795270a8a0177d9af27498091a6821a1ed6fb1/app/db/async_blocker.py#L30-L31

This is the mypy error:
```
app/db/async_blocker.py:30: error: "datetime" has no attribute "set"
app/db/async_blocker.py:30: error: Unsupported left operand type for | ("datetime")
app/db/async_blocker.py:31: error: "datetime" has no attribute "set"
```

Reply:
Are you adding pynamodb to your mypy dependencies?

Reply:
when adding `pynamodb` to  `additional_dependencies` the error changed to

```
app/db/async_blocker.py:30: error: Unsupported left operand type for | ("UTCDateTimeAttribute")
```

Reply:
The culprit is that PynamoDB's `Attribute` type stub doesn't define
```
def __or__() -> pynamodb.expressions.operand._IfNotExists: ...
```
and similarly `_IfNotExists` isn't defined in operand.pyi.

cc @garrettheel